lib/document/stateTransition/applyDocumentsBatchTransitionFactory.js

```diff
@@ -28,11 +28,13 @@ function applyDocumentsBatchTransitionFactory(
    * @return {Promise<*>}
    */
   async function applyDocumentsBatchTransition(stateTransition) {
+    const dataContractId = stateTransition.getTransitions()[0].getDataContractId();
```

Comment:
Aren't we suppose to have different documents for different contracts?

Reply:
Not right now, but in the future when we figure out atomic operations between contracts. We just changed the structure in order to support it.

Reply:
We'll have an error that says that we're trying to validate against the wrong contract, but since this behavior isn't specified anywhere, it can be really confusing for the developer. I'm not sure what's the best approach would be. Actually I don't see any reason why we don't want to allow documents for different contracts in the same st. This probably won't happen much now, but I can imagine a use-case where the app uses two contracts at the same time, especially since currently we can't update the contract, and the only way to add new functionality to the app is to create another contract

Reply:
Yeah, we could allow people to put docs from different data contracts now but it will lead to significant changes in validation functions. I thought that we could postpone these changes for the next release.